From: Michael Tokarev <mjt@tls.msk.ru>
Subject: fix alignment checks
Date: Sat, 01 Nov 2025 14:48:45 +0300

Current code assumes the same alignment of fields on
all architectures, while the actual alignment depends
on the architecture-specific alignment of particular
fields.

Signed-off-by: Michael Tokarev <mjt@tls.msk.ru>

--- a/src/linux/aio.rs
+++ b/src/linux/aio.rs
@@ -346,3 +343,3 @@ mod test {
             ::std::mem::align_of::<IoEvent>(),
-            8usize,
+            ::std::mem::align_of::<u64>(),
             concat!("Alignment of", stringify!(IoEvent))
@@ -360,3 +357,3 @@ mod test {
             ::std::mem::align_of::<IoControlBlock>(),
-            8usize,
+            ::std::mem::align_of::<u64>(),
             concat!("Alignment of", stringify!(IoControlBlock))
